liquidprompt

```diff
@@ -991,6 +991,71 @@ _lp_grep_fields() {
     done <"$file"
 }
 
+# Adds as much $3 character (e.g. spaces) between $1 and $2,
+# so as to make the resulting string the same width as the current terminal.
+_lp_fill() {
+
+    local left="$1" right="$2" fillchars="${3:-" "}" splitends="${4:-1}"
+
+    # Compute the gap between left and right.
+    local ret
+    __lp_strip_escapes "$left"
+    local left_as_text="$ret"
+    __lp_strip_escapes "$right"
+    local right_as_text="$ret"
+    local gap_width=$((${COLUMNS:-80}-${#left_as_text}-${#right_as_text}))
+    if [[ $gap_width -lt 0 ]] ; then
+        # FIXME this would be larger than the actual width.
+        # Maybe think of an option to reduce either the right or left side?
+        # Or insert a newline?
+        lp_fill="${left}${right}"
+        return
```

Comment:
I have a solution: `return 1` and leave it up to the theme to decide what to do.

The theme can then either:
* Throw away the return string, and try to shorten one side or the other and try again, or
* Ignore the return value, and use the return string even though it is too long.